### PR TITLE
tests: simplified Makefiles for device driver test apps

### PR DESCRIPTION
**Pull request by haukepetersen**
@cgundogan pointed out a nicer way to specify the peripheral configuration for out device drivers tests to me, so I went ahead and cleaned up those tests... I think they look way nicer now and should be easier to maintain in the future...

### CODE REVIEW

File: tests/driver_at86rf2xx/Makefile [hunk 1]
```diff
@@ -3,71 +3,56 @@ include ../Makefile.tests_common
...
 ifneq (true,$(USE_BOARD_PARAMETERS))
-  # This adds . to include path so generic at86rf2xx_params.h gets picked
-  # up.  All boards actually having such a device on board should define
-  # USE_BOARD_PARAMETERS=true above to skip this step, as the board provides
-  # this header.
-  CFLAGS += -I$(CURDIR)
 
-ifneq (,$(ATRF_SPI))
+  # set default device parameters in case they are undefined
+  DRIVER ?       ?= at86rf231
```
Comment (cgundogan) [inline tests/driver_at86rf2xx/Makefile]: there is one extra `?`

Comment (haukepetersen) [inline tests/driver_at86rf2xx/Makefile]: ups, must have slipped through...

**Comment by haukepetersen** [comment]
fixed and squashed

**Comment by haukepetersen** [comment]
Put Travis on it...

**Comment by cgundogan** [comment]
Travis fails with for group `cortex_m3_1`
```
RIOT-OS/RIOT/sys/auto_init/netif/auto_init_at86rf2xx.c:27:30: fatal error:
at86rf2xx_params.h: No such file or directory
```

**Comment by haukepetersen** [comment]
should be fixed now.

**Comment by cgundogan** [comment]
Travis is green - ACK and GO

**Pull request merged**